src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -649,87 +635,105 @@ private void AddTransactionToWallet(Transaction transaction, TxOut utxo, int? bl
-                            break;
-                        case TxOutType.TX_NONSTANDARD:
-                        case TxOutType.TX_SCRIPTHASH:
-                            destinationAddress = paidToOutput.ScriptPubKey.GetDestinationAddress(this.network).ToString();
-                            break;
-                        case TxOutType.TX_MULTISIG:
-                        case TxOutType.TX_NULL_DATA:
-                        case TxOutType.TX_SEGWIT:
-                            break;
-                    }
+                spentTransaction.SpendingDetails = this.BuildSpendingDetails(transaction, paidToOutputs, blockHeight, blockHash, block, withdrawal);
+            }
+            else if (spentTransaction.SpendingDetails.BlockHeight == null)
+            {
+                // 2) If there are unconfirmed existing spending details, always overwrite with new one. Could be a
+                //   "more" signed tx, a FullySigned mempool tx or a confirmed block tx.
 
-                    payments.Add(new PaymentDetails
-                    {
-                        DestinationScriptPubKey = paidToOutput.ScriptPubKey,
-                        DestinationAddress = destinationAddress,
-                        Amount = paidToOutput.Value
-                    });
-                }
+                this.logger.LogTrace("Spending UTXO '{0}-{1}' is being overwritten. BlockHeight={2}", spendingTransactionId, spendingTransactionIndex, blockHeight);
 
-                SpendingDetails spendingDetails = new SpendingDetails
-                {
-                    TransactionId = transaction.GetHash(),
-                    Payments = payments,
-                    CreationTime = DateTimeOffset.FromUnixTimeSeconds(block?.Header.Time ?? transaction.Time),
-                    BlockHeight = blockHeight,
-                    BlockHash = blockHash,
-                    Hex = transaction.ToHex(),
-                    IsCoinStake = transaction.IsCoinStake == false ? (bool?)null : true
-                };
+                spentTransaction.SpendingDetails = this.BuildSpendingDetails(transaction, paidToOutputs, blockHeight, blockHash, block, withdrawal);
+            }
+            else if (spentTransaction.SpendingDetails.BlockHeight != null && blockHeight == null)
+            {
+                // 3) If we have confirmed existing spending details, and this is coming in unconfirmed,
+                //   probably just unlucky concurrency issues, e.g. tx from mempool coming in after confirmed in a block.
+                this.logger.LogTrace("Unconfirmed spending UTXO '{0}-{1}' is being ignored. Already confirmed in block.", spendingTransactionId, spendingTransactionIndex);
+            }
+            else
+            {
+                // 4) If we have confirmed existing spending details, and this is also coming in confirmed, then something has gone wrong.
```

Comment:
This can also happen if the `WithdrawalDetails` are `null` which would lead to `FindWithdawalTransactions` failing and a subsequent `ProcessTransaction` call from the CCTS `Start` code.